activerecord/test/cases/relation/or_test.rb

```diff
@@ -139,6 +139,14 @@ def test_or_with_scope_on_association
       end
     end
 
+    def test_or_with_annotate
+      quoted_posts = Post.quoted_table_name
```

Comment:
The `quoted_table_name` should be escaped by `Regexp.escape`, since it is qouted as `[posts]` in SQLServer.

```suggestion
      quoted_posts = Regexp.escape(Post.quoted_table_name)
```

Reply:
Thanks for your review. I updated.